### PR TITLE
Rework the profile to allow the setup of multiple OpenIO repositories…

### PR DESCRIPTION
**Pull request by vincent-legoll**
… at once

- Use a dict of dicts to select the OpenIO repositories
  configurations to setup
- Do not use -present & -absent tasks files, combine them &
  only use "{{ repo.state }}"
- Do not use repository creds for GPG key, as it is located
  outside of the password-protected area

### CODE REVIEW

File: vars/main.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 ---
 # vars file for ansible-role-openio-repository
-openio_repository_url: "http://{{ openio_repository_creds }}mirror.openio.io/pub/repo/openio"
+
+# Credentials for private repo, dependent on "with_dict" loop variable: "item"
```
Comment (cdelgehier) [inline vars/main.yml]: Can you explain ?

File: tasks/Ubuntu.yml [hunk 2]
```diff
@@ -38,7 +40,7 @@
 
 - name: "Update APT cache"
   apt:
-    update_cache: yes
+    update_cache: "yes"
```
Comment (cdelgehier) [inline tasks/Ubuntu.yml]: It's not `"yes"` but `true`
`update_cache=dict(aliases=['update-cache'], type='bool'),`

**Comment by cdelgehier** [comment]
Can you make a PR for the CI branch and commit a change in the README.md ?

**Review by cdelgehier — CHANGES_REQUESTED** [review]
I would like to see  a playbook  with this role, because Travis does not seem happy
The lint is new but do not play it
I like the idea of joining the 2 files (...-state.yml)

File: tasks/Ubuntu.yml (hunk 2 above)
Comment (vincent-legoll) [reply tasks/Ubuntu.yml]: I only added the quotes, but OK I'll fix it

File: vars/main.yml (hunk 1 above)
Comment (vincent-legoll) [reply vars/main.yml]: This variable, using "{{ item.something }}" is meant to be used in a "with_dict" loop... See its usage in the tasks files

**Comment by vincent-legoll** [comment]
In fact I should have put a WIP in the PR name, but yes I'll do the CI branch & README too. I only wanted to get travis testing done yesterday evening... Which already caught some defects.

**Comment by vincent-legoll** [comment]
@cdelgehier : This is still WIP (wrt the README) but should be testable.

**Pull request merged**